src/main/java/pl/javastart/lista/TaskController.java

```diff
@@ -0,0 +1,80 @@
+package pl.javastart.lista;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.util.List;
+import java.util.Optional;
+
+@Controller
+public class TaskController {
+
+    private TaskRepository taskRepository;
+
+    public TaskController(TaskRepository taskRepository) {
+        this.taskRepository = taskRepository;
+    }
+
+    @GetMapping("/")
+    public String home(Model model, @RequestParam(required = false) Category category) {
+        List<Task> taskList;
+        if (category != null) {
+            taskList = taskRepository.findByCategory(category);
+        } else {
+            taskList = taskRepository.findAll();
+        }
+        model.addAttribute("taskList", taskList);
+        return "home";
+    }
+
+    @GetMapping("/archive")
+    public String archive(Model model, @RequestParam(required = false) Task isDone) {
```

Comment:
Przyp próbie wejścia do archiwum leci mi `NullPointerException`. Wydaje mi się, że `isDone` powinno być tu booleanem, tak sugeruje jego nazwa :P Aczkolwiek jako, że jest to archiwum to mogłoby nie być tu tego request parama i po prostu przekaywać do etody repoytorium zawsze `true`, jako że takie zadania są zarchwizowane. Chociaż najelpzsą opcją wydaje mi się przenieść ten request param do metody wyżej, i zaimlementować bowyższą metodę, tak by uwzględniała oba parametry ;)